webscraping/webscrape.py

```diff
@@ -32,16 +35,51 @@ def requestHTML(reqName):
         file.close()
     return OUTPUT
 
-for idx in range(30):
+
+def parseExercise(idx):
     OUTPUT = requestHTML("/resources/everyone/exercise-library/" + str(idx) + "/")
     linkIndexes = [i for i in range(len(OUTPUT)) if OUTPUT.startswith('"', i)]
     if len(linkIndexes) > 1: 
-        print(OUTPUT[linkIndexes[0] + 1:linkIndexes[1]])
         newLink = OUTPUT[linkIndexes[0] + 1:linkIndexes[1]]
         exerciseHTML = requestHTML(newLink)
-        #From here all of the information I want on a given exercise is on exerciseHTML
-        #Things I can grab, Exercise Title, Body parts, equipment, instructions, relative difficulty
-        exerciseTitle = firstBetween(exerciseHTML,'exercise-hero__title">', '<' )
-        exerciseBodyPart = firstBetween(exerciseHTML, '<dd>', '<')
-        if exerciseTitle and exerciseBodyPart:
-            print(exerciseTitle+  " " + exerciseBodyPart)
\ No newline at end of file
+        
+        exerciseTitle = find_between(exerciseHTML,'exercise-hero__title">', '<' )
+        exerciseBodyPart = find_between(exerciseHTML, '<dd>', '<')
+        exerciseEquipment = find_between(exerciseHTML, 'Equipment:</dt>\\r\\n\\t\\t<dd>', '</dd>')
+        exerciseImage = find_between(exerciseHTML, "<div class=\"exercise-hero__image\" style=\"background-image: url(\\'", "\\'")
+        steps = []
+
+        if exerciseTitle and exerciseBodyPart and exerciseEquipment:
+            stepNumber = 1
+
+            while ("<h2>Step " + str(stepNumber)) in exerciseHTML:
+                currentStep = find_between(exerciseHTML, "<h2>Step " + str(stepNumber) + "</h2>\\r\\n<p>", "<")
+                if not currentStep:
+                    currentStep = find_between(exerciseHTML, "<h2>Step " + str(stepNumber) + "&gt;</h2>\\r\\n<p>", "<") #alternate step format, has >  character after Step #
+                steps.append(currentStep)
+                stepNumber = stepNumber + 1
+
+            return {
+                "name" : exerciseTitle,
+                "bodyParts" : exerciseBodyPart,
+                "equipment" : exerciseEquipment,
+                "steps" : steps,
+                "image" : exerciseImage
+                }
+    return None    
+
+exercises = []
+
+for idx in range(400):
+    exercise = parseExercise(idx)
+    if exercise:
+        exercises.append(exercise)
+
+
+dictionary = {
+    "exercises" : exercises
+}
+json_object = json.dumps(dictionary, indent=4)
+
+with open("exercises.json", "w") as outfile:
+    outfile.write(json_object)
```

Comment:
For better formatting, maybe something like this

```
def main():
    exercises = []
    for idx in range(400):
        exercise = parse_exercise(idx)
        if exercise:
            exercises.append(exercise)
    with open("exercises.json", "w") as outfile:
        json.dump({"exercises": exercises}, outfile, indent=4)
if __name__ == "__main__":
    main()
```

Reply:
What does if __name__ == "__main__": do in this instance?